2-mandatory/4-stocks.js

```diff
@@ -64,31 +81,54 @@ function getPriceChanges(closingPricesForAllStocks) {
     The price should be shown with exactly 2 decimal places.
 */
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
-    // TODO
-}
+  let ArraySentenceStock = [];
+  let highestStock = [];
+  let stockNameCapitalised = [];
+
+  for (let stockPrice of closingPricesForAllStocks) {
+    let maxStockPrice = 0;
+    for (let compareStockPrice of stockPrice) {
+      if (compareStockPrice > maxStockPrice) {
+        maxStockPrice = compareStockPrice;
+      }
+    }
+
+    highestStock.push(maxStockPrice.toFixed(2));
+  }
 
+  for (let stockName of stocks) {
+    stockNameCapitalised.push(stockName.toUpperCase());
+  }
+
+  for (let i = 0; i < stocks.length; i++) {
+    let sentence = `The highest price of ${stockNameCapitalised[i]} in the last 5 days was ${highestStock[i]}`;
+    ArraySentenceStock.push(sentence);
+  }
+
+  return ArraySentenceStock;
+}
```

Comment:
All these methods work well and your code is understandable - this is a difficult exercise, so well done.

Reply:
Thank you